ctapipe/reco/energy_regressor.py

```diff
@@ -0,0 +1,383 @@
+                except AttributeError:
+                    # in case the energy is not given as an astropy quantity
+                    # let's hope that the user keeps proper track of the unit themself
+                    trainTarget[cam_id] += [en]*len(tels)
+        return trainFeatures, trainTarget
+
+    def fit(self, X, y):
+        """
+        This function fits a model against the collected features; separately for every
+        telescope identifier.
+
+        Parameters
+        ----------
+        X : dictionary of lists of lists
+            Dictionary that maps the telescope identifiers to lists of feature-lists.
+            The values of the dictionary are the lists `scikit-learn` regressors train on
+            and are supposed to comply to their format requirements
+            e.g. each featuer-list has to contain the same features at the same position
+        y : dictionary of lists
+            the energies corresponding to all the feature-lists of `X`
+
+        Returns
+        -------
+        self
+
+        Raises
+        ------
+        KeyError
+            in case `X` contains keys that are either not in `y` were not provided before
+            with `cam_id_list`.
+
+        """
+        for cam_id in X:
+            if cam_id not in y:
+                raise KeyError("cam_id '{}' in X but not in y: {}"
+                               .format([k for k in y]))
+
+            if cam_id not in self.reg_dict:
+                raise KeyError("cam_id '{}' in X but no regressor defined: {}"
+                               .format(cam_id, [k for k in self.reg_dict]))
+
+            # for every `cam_id` train one regressor
+            self.reg_dict[cam_id].fit(X[cam_id], y[cam_id])
+
+        return self
+
+    def predict(self, X, cam_id=None):
+        """
+        In the tradition of scikit-learn, `.predict` takes a "list of feature-lists" and
+        returns an estimate for targeted quantity for every  set of features.
+
+        Parameters
+        ----------
+        X : list of lists of floats
+            the list of feature-lists
+        cam_id : any (e.g. string, int), optional
+            identifier of the singular camera type to consider here
+            if not set, `.reg_dict` is assumed to have a single key which is used in place
+
+        Returns
+        -------
+        predict : list of floats
+            predictions for the target quantity for every set of features given in `X`
+
+        Raises
+        ------
+        ValueError
+            if `cam_id is None` and the number of registered regressors is not 1
+        """
+
+        if cam_id is None:
+            if len(self.reg_dict) == 1:
+                cam_id = next(iter(self.reg_dict.keys()))
+            else:
+                raise ValueError("you need to provide a cam_id")
+
+        return self.reg_dict[cam_id].predict(X)
+
+    def predict_dict(self, X):
```

Comment:
Maybe find a better name, like `predict_for_telescopes`

Also, not only the mean of the predicitions might be interesting, but also the std dev or other measures of centricity like the median.